source/Calamari/Kubernetes/Conventions/GatherAndApplyRawYamlConvention.cs

```diff
@@ -0,0 +1,172 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Calamari.Common.Commands;
+using Calamari.Common.Features.Processes;
+using Calamari.Common.Plumbing.Extensions;
+using Calamari.Common.Plumbing.FileSystem;
+using Calamari.Common.Plumbing.Logging;
+using Calamari.Common.Plumbing.Variables;
+using Calamari.Deployment.Conventions;
+using Calamari.Kubernetes.Integration;
+using Newtonsoft.Json.Linq;
+using Octopus.CoreUtilities.Extensions;
+
+namespace Calamari.Kubernetes.Conventions
+{
+    public class GatherAndApplyRawYamlConvention: IInstallConvention
```

Comment:
Should we have tests around this and the other new conventions?

Reply:
This is the main one that needs testing which I was planning to do later as described in my other comment (https://github.com/OctopusDeploy/Calamari/pull/1010/files#r1178673725)

`KubernetesAuthContextConvention` basically just wraps the `SetupKubectlAuthentication` class which is tested via the current Kubernetes tests.

`ResourceStatusReportConvention` basically just wraps `ResourceStatusReportExecutor` which was split out from `ResourceStatusReportScriptWrapper` which has tests that Yihao wrote. 

These tests will need to be modified when we move all kubernetes steps to their own Calamari Commands (instead of being scripts) which means we will be able to delete the script wrapper classes.